### PR TITLE
[C#] Allow users to quickly populate native string buffers with utf8 bytes

### PR DESCRIPTION
**Pull request by yuslepukhin**
### Description
Introduce an API that allows users to gain access to a string tensor element buffer of requested length in bytes
so then can quickly load any utf8 data.

### Motivation and Context
Useful for testing an otherwise.

### CODE REVIEW

File: csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs [hunk 1]
```diff
@@ -261,6 +261,23 @@ public ReadOnlySpan<byte> GetStringElementAsSpan(int index)
...
         }
 
+        /// <summary>
+        /// This API resizes String Tensor element to the requested amount of bytes (UTF-8)
+        /// and returns a span to the resized buffer. This API is useful when you have UTF-8 data
+        /// that you want to quickly stick into the OrtValue native memory.
+        /// </summary>
+        /// <param name="bytesCount">requested buffer size in bytes</param>
+        /// <param name="index">flat index of the element in the string tensor</param>
+        /// <returns></returns>
```
Comment (edgchen1) [inline csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs:271]: add return value description?

File: csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs [hunk 2]
```diff
@@ -261,6 +261,23 @@ public ReadOnlySpan<byte> GetStringElementAsSpan(int index)
...
+        public Span<byte> GetMutableStringElementAsSpan(int bytesCount, int index)
```
Comment (edgchen1) [inline csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs:272]: I wonder if we should instead have another `FillStringTensorElement` overload which takes utf-8 bytes?

**Review by pranavsharma — COMMENTED** [review]
LGTM other than the comment on filling the return value desc.

File: csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs (hunk 2 above)
Comment (yuslepukhin) [reply csharp/src/Microsoft.ML.OnnxRuntime/OrtValue.shared.cs]: Fixed.

**Review by edgchen1 — APPROVED** [review]

**Pull request merged**